### DISCUSSION TITLE
Battery status

### DISCUSSION THREAD

**Discussion by malcerz** [gopro/labs#451] [Feature Requests]
Would it be possible to add the battery level of the camera to the metadata?

**Comment by dnewman-gpsw** [comment]
It is possible to script that on HERO11.

![image](https://github.com/gopro/labs/assets/10856084/2cf77f22-50cc-4152-a45e-8f7b30423bdb)

This will add global metadata "BATT" and store the last battery level for each capture.

**Reply by malcerz** [reply]
I loaded the code, the camera screen showed confirmation, but I don't see BATT data in the stream
goprotelemetryextractor only sees this data:
![image](https://github.com/gopro/labs/assets/55062569/1aad9526-8280-4840-a0e5-0d50814b8bec)

**Reply by dnewman-gpsw** [reply]
That tool is looking in the wrong place.  It is in header, not the track data.  Use https://gopro.github.io/labs/control/metadata/

**Reply by malcerz** [reply]
I need just for this tool - for the Telemetry Overlay program. Is there a way to put battery data in a place where Telemetry Overlay can display it?

**Reply by dnewman-gpsw** [reply]
You need to ask them, that is not a tool provided by GoPro.  However, this would only have one battery value per video clip as the above script can only store the last battery level.

**Reply by malcerz** [reply]
So it's just one last battery record?
I'm looking for a way to save all battery level changes from turning the camera on to turning it off, so that I can include it in the video.
I'm making a material with an overlay containing a lot of data and I'm only missing the battery charge level. I managed to get the battery level from the Garmin watch I use to record the GPS track, and only GoPro hides this information from me
It can't be output to mp4 as data, for example, about the ISO of the camera or from the accelerometer that the original soft provides as standard?
https://youtu.be/a_ejdUFsaBY
![image](https://github.com/gopro/labs/assets/55062569/8e860b26-1950-493b-bd39-c5c35b128278)

**Reply by dnewman-gpsw** [reply]
No possible currently to continuously record battery level.
